gluten-core/src/main/scala/io/glutenproject/extension/GlutenPlan.scala

```diff
@@ -52,41 +67,45 @@ trait GlutenPlan extends SparkPlan with LogLevelUtil {
     try {
       TransformerState.enterValidation
       val res = doValidateInternal()
-      if (!res.validated) {
+      if (!res.isValid) {
         TestStats.addFallBackClassName(this.getClass.toString)
-        // the reason must be set if failed to validate
-        assert(res.reason.isDefined)
       }
       res
     } catch {
       case e: Throwable =>
         TestStats.addFallBackClassName(this.getClass.toString)
-        logValidateFailure(s"Validation failed with exception for plan: $nodeName, due to:", e)
-        notOk(e.getMessage)
+        logValidationMessage(s"Validation failed with exception for plan: $nodeName, due to:", e)
+        ValidationResult.notOk(e.getMessage)
     } finally {
       TransformerState.finishValidation
     }
   }
 
   protected def doValidateInternal(): ValidationResult = ValidationResult.ok
 
-  protected def logValidateFailure(msg: => String, e: Throwable): Unit = {
-    if (printStackOnValidateFailure) {
-      logOnLevel(validateFailureLogLevel, msg, e)
+  protected def doNativeValidation(context: SubstraitContext, node: RelNode): ValidationResult = {
+    if (node != null && enableNativeValidation) {
+      val planNode = PlanBuilder.makePlan(context, Lists.newArrayList(node))
+      // TODO Unify doValidate interface, remove this branch
+      if (BackendsApiManager.chBackend &&
+        BackendsApiManager.getValidatorApiInstance.doValidate(planNode)) {
```

Comment:
It may be better to move `BackendsApiManager.getValidatorApiInstance.doValidate(planNode)` into this `if` branch and return ok or notOk accordingly. Right?